Sources/Testing/SourceAttribution/SourceLocation.swift

```diff
@@ -12,14 +12,19 @@
 public struct SourceLocation: Sendable {
   /// The file ID of the source file.
   ///
+  /// - Precondition: The value of this property must not be empty and be
+  ///   formatted as described in the documentation for the
+  ///   [`#fileID`](https://developer.apple.com/documentation/swift/fileID()).
+  ///   macro in the Swift standard library.
+  ///
   /// ## See Also
   ///
   /// - ``moduleName``
   /// - ``fileName``
   public var fileID: String {
-    didSet {
-      precondition(!fileID.isEmpty)
-      precondition(fileID.contains("/"))
+    willSet {
+      precondition(!newValue.isEmpty, "SourceLocation.fileID must not be empty (was \(newValue))")
+      precondition(newValue.contains("/"), "SourceLocation.fileID must use the format described in documentation for #fileID (was \(newValue))")
```

Comment:
Nitpick (feel free to disregard) but we can probably just say `"SourceLocation.fileID must be a well-formed file ID (was \(newValue))"`. No need to call out the docs here. (Just a thought on style.)